zcash_primitives/src/transaction/components/tze.rs

```diff
@@ -117,15 +117,15 @@ impl TzeIn<<Authorized as Authorization>::Witness> {
     pub fn read<R: Read>(mut reader: &mut R) -> io::Result<Self> {
         let prevout = OutPoint::read(&mut reader)?;
 
-        let extension_id = CompactSize::read(&mut reader)?;
-        let mode = CompactSize::read(&mut reader)?;
+        let extension_id: u32 = CompactSize::read_t(&mut reader)?;
+        let mode: u32 = CompactSize::read_t(&mut reader)?;
```

Comment:
I'm pretty sure the compiler can figure out these types for you:
```suggestion
        let extension_id = CompactSize::read_t(&mut reader)?;
        let mode = CompactSize::read_t(&mut reader)?;
```